patches/server/0994-Fix-rotation-when-spawning-entities.patch

```diff
@@ -0,0 +1,40 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Gameoholic <viner.atmic@gmail.com>
+Date: Sun, 30 Jul 2023 13:30:34 +0300
+Subject: [PATCH] Fix rotation when spawning entities
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftRegionAccessor.java b/src/main/java/org/bukkit/craftbukkit/CraftRegionAccessor.java
+index 995b3e31c5725f3d8879df349351eae3bcdfdddf..2f028a771e79c249ac511ecb69916f6e07beef58 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftRegionAccessor.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftRegionAccessor.java
+@@ -971,10 +971,8 @@ public abstract class CraftRegionAccessor implements RegionAccessor {
+             entity = new net.minecraft.world.entity.projectile.EvokerFangs(world, x, y, z, (float) Math.toRadians(yaw), 0, null);
+         } else if (Marker.class.isAssignableFrom(clazz)) {
+             entity = net.minecraft.world.entity.EntityType.MARKER.create(world);
+-            entity.setPos(x, y, z);
+         } else if (Interaction.class.isAssignableFrom(clazz)) {
+             entity = net.minecraft.world.entity.EntityType.INTERACTION.create(world);
+-            entity.setPos(x, y, z);
+         } else if (Display.class.isAssignableFrom(clazz)) {
+             if (BlockDisplay.class.isAssignableFrom(clazz)) {
+                 entity = net.minecraft.world.entity.EntityType.BLOCK_DISPLAY.create(world);
+@@ -983,13 +981,14 @@ public abstract class CraftRegionAccessor implements RegionAccessor {
+             } else if (TextDisplay.class.isAssignableFrom(clazz)) {
+                 entity = net.minecraft.world.entity.EntityType.TEXT_DISPLAY.create(world);
+             }
+-
+-            if (entity != null) {
+-                entity.setPos(x, y, z);
+-            }
+         }
+ 
+         if (entity != null) {
++            // Paper Start - Update position and rotation
++            entity.setPos(x, y, z);
```

Comment:
Don't move the setPos here. Each entity already has it's own logic for settings its position, its at best redundant, and at worst breaking whatever logic behavior might happen above to set the position.

Reply:
I mean I thought it was because there's a ton of extra logic in moveTo that isn't required for setting an initial position.

Reply:
I did that because Owen suggested to make a fix that makes all entities respect yaw/pitch, but you're right. In that case, can I revert back to my original commit, which just fixed the rotation for display entities?

Reply:
Probably. It's out of scope of this PR, can be a future thing to cleanup.